src/thing_shots.c

```diff
@@ -767,14 +768,36 @@ TbBool shot_kill_creature(struct Thing *shotng, struct Thing *creatng)
 
 long melee_shot_hit_creature_at(struct Thing *shotng, struct Thing *trgtng, struct Coord3d *pos)
 {
+    struct Thing* shooter = INVALID_THING;
+    if (shotng->parent_idx != shotng->index)
+    {
+        shooter = thing_get(shotng->parent_idx);
+    }
+    if (!thing_is_invalid(shooter))
+    {
+        if (is_thing_directly_controlled(shooter))
+        {
+            if (first_person_dig_claim_mode)
+            {
+                if ( (shotng->model == ShM_SwingFist) || (shotng->model == ShM_SwingSword) )
+                {
+                    if (shooter->model == 20)
```

Comment:
Linking this to the model number is not a very elegant solution. It would be much nicer to to add a new creature property and link it to that instead. Like these:
`Properties = BLEEDS HUMANOID_SKELETON `